inc/impact.class.php

```diff
@@ -0,0 +1,1091 @@
+<?php
+
+if (!defined('GLPI_ROOT')) {
+   die("Sorry. You can't access this file directly");
+}
+
+/**
+ * @since 9.5.0
+ */
+class Impact extends CommonGLPI {
+   // Constants used to express the direction or "flow" of a graph
+   // Theses constants can also be used to express if and edge is reachable
+   // when exploring the graph forward, backward or both (0b11)
+   const DIRECTION_FORWARD    = 0b01;
+   const DIRECTION_BACKWARD   = 0b10;
+
+   // Default colors used for the edges of the graph according to their flow
+   const DEFAULT_COLOR            = 'black';  // No flow, this mean the edge is not accessible from the starting point of the graph
+   const IMPACT_COLOR             = '#ff3418'; // Forward
+   const DEPENDS_COLOR            = '#1c76ff'; // Backward
+   const IMPACT_AND_DEPENDS_COLOR = '#ca29ff'; // Forward and backward
+
+   public static function getTypeName($nb = 0) {
+      return _n('Asset impact', 'Asset impacts', $nb);
+   }
+
+   public function getTabNameForItem(CommonGLPI $item, $withtemplate = 0) {
+      global $CFG_GLPI, $DB;
+
+      // Class of the current item
+      $class = get_class($item);
+
+      // Only enabled for CommonDBTM
+      if (!is_a($item, "CommonDBTM")) {
+         throw new InvalidArgumentException(
+            "Argument \$item ($class) must be a CommonDBTM)."
+         );
+      }
+
+      // If on an asset, get the number its direct dependencies
+      if (isset($CFG_GLPI['impact_asset_types'][$class])) {
+         $total = count($DB->request([
```

Comment:
As far as I know, counting should only be done if the configuration is enabled

Reply:
I'm not sure to understand, isn't it already the case as we are checking `$CFG_GLPI['impact_asset_types'][$class]`  ?

Reply:
I'm talking about `$_SESSION['glpishow_count_on_tabs']`

Reply:
Fixed in latest commit.